File: Data Structures & Algorithms/max-water-container/submission-2.java
class Solution {
    public int maxArea(int[] heights) {
        int max=0;
        int left=0;
        int right=heights.length-1;
        while(left<right){
            int maxheight=(right-left)*Math.min(heights[left],heights[right]);
            max=Math.max(max,maxheight);
            if (heights[left] < heights[right]) {
                left++;
            } else {
                right--;
            }
        }
        
        return max;
        
    }
}

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-2.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int l=0;
        int r=1;
        
        int maxp=0;
        while(r<prices.size()){
            if(prices[l]<prices[r]){
                int tp=prices[r]-prices[l];
                maxp=max(maxp,tp);
                
            }else{
                l=r;
            }
            r++;
        }
        return maxp;
        
    }
};
